lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+        if current_user:
+            current_user_id = current_user.id
+        else:
+            current_user_id = None
+
+        value = post_data.get("value")
+        if not value:
+            self.logger.warning(
+                "value not provided during metadata creation. user_id: {}",
+                current_user_id,
+            )
+            abort(400, message="Value field missing")
+
+        try:
+            # Create a new metadata session
+            metadata_session = Metadata(
+                value=value, key=metadata_key, user_id=current_user_id
+            )
+            # insert the metadata session for a user
+            metadata_session.add()
+            self.logger.info(
+                "New metadata session created for user_id {}", current_user_id
+            )
+        except Exception:
+            self.logger.exception("Exception occurred during the Metadata creation")
+            abort(500, message="INTERNAL ERROR")
+        else:
+            response_object = {
+                "message": "success",
+                "data": metadata_session.get_json(),
+            }
+            return make_response(jsonify(response_object), 201)
+
+    @Auth.token_auth.login_required(optional=True)
+    def get(self, key):
+        """
+        Get request for querying all the metadata sessions for a user.
+        returns the list of all the metadata sessions associated with a logged in user.
+        This requires a Pbench auth token in the header field
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
+                    "data": {
+                        "sessions": [
+                            {"id": "metadata_id",
+                            "value": "client text blob",
+                            "key": "key string",
+                            "created": "datetime string",
+                            "updated": "datetime string", },
+                            {}]
+                        }
+                }
+        """
+        if not key:
+            self.logger.warning("Metadata key not provided during metadata query")
+            abort(400, message="Missing metadata key in the query url")
+
+        current_user = self.auth.token_auth.current_user()
+        if current_user:
+            current_user_id = current_user.id
+        else:
+            current_user_id = None
+        print(current_user_id)
+        try:
+            # Fetch the metadata sessions
+            # If the key is favorite, we return only favorite sessions,
+            # else we return all the saved and favorite sessions
+            if key.upper() == "FAVORITE":
+                metadata_sessions = Metadata.query(
+                    user_id=current_user_id, key=MetadataKeys[key.upper()].value
```

Comment:
If you're going to always use the uppercased key (I'd prefer lowercased since uppercase looks like __SHOUTING__), it should be done once at the start instead of on each reference; that's error-prone.

Reply:
I updated the code so the server can accept any user defined key (and not restrict to saved/favorite) and we will keep it lowercase in our database. I also got rid of Enums and kept the key column as a user-defined string